pandas/core/reshape.py

```diff
@@ -410,13 +413,24 @@ def unstack(obj, level, fill_value=None):
 
     if isinstance(obj, DataFrame):
         if isinstance(obj.index, MultiIndex):
-            return _unstack_frame(obj, level, fill_value=fill_value)
+            unstacked = _unstack_frame(obj, level, fill_value=fill_value)
+        else:
+            unstacked = obj.T.stack(dropna=False)
+
+        if len(unstacked.shape) == 1:
+            return obj._constructor_sliced(unstacked)
         else:
-            return obj.T.stack(dropna=False)
+            return obj._constructor(unstacked)
+
     else:
         unstacker = _Unstacker(obj.values, obj.index, level=level,
                                fill_value=fill_value)
-        return unstacker.get_result()
+        unstacked = unstacker.get_result()
+
```

Comment:
same here, this now has logic in 2 places, the creation itself of the unstacked result, then the return class. should be a part of ``get_result`` (it certainly could be another method that is *called* by ``get_result``) though